scylla/tests/conftest.py

```diff
@@ -0,0 +1,46 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import os
+
+import mock
+import pytest
+
+from datadog_checks.dev import docker_run
+from datadog_checks.utils.common import get_docker_hostname
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+HOST = get_docker_hostname()
+INSTANCE_PORT = 9180
+INSTANCE_URL = "http://{}:{}/metrics".format(HOST, INSTANCE_PORT)
+
+
+@pytest.fixture(scope='session')
+def dd_environment():
+    compose_file = os.path.join(HERE, 'compose', 'docker-compose.yaml')
+
+    with docker_run(compose_file, log_patterns=[r'init - Scylla version \S* initialization completed.']):
+        instances = {'instances': [{'instance_endpoint': INSTANCE_URL}]}
+        yield instances
+
+
+@pytest.fixture(scope="session")
+def db_instance():
+    return {'instance_endpoint': INSTANCE_URL, 'tags': ['instance_test']}
+
+
+@pytest.fixture()
+def mock_db_data():
+    f_name = os.path.join(os.path.dirname(__file__), 'fixtures', 'scylla_metrics.txt')
+    with open(f_name, 'r') as f:
+        text_data = f.read()
+    with mock.patch(
+        'requests.get',
+        return_value=mock.MagicMock(
+            status_code=200,
+            raise_for_status=lambda **kwargs: kwargs,
```

Comment:
Where do you need this?

Reply:
There was an earlier iteration where `mock.Patch` used `new` vs `return_value`, and in that case the object needed to handle `raise_for_status`.

Since this went back to `return_value` extra arg can be removed.

Reply:
🤔 Not sure I understand, the object returned by mock.patch will be of type `MagicMock`.
Which means that any attributes or method of that object will be no-op. Isn't that enough?

Reply:
I thought so too, but something gets handled different when it's passed via `new` like so:

```python
def mock_get(*args, **kwargs):
    url = args[0]
    data = mocked_metrics(url)

    return mock.MagicMock(
        status_code=200,
        raise_for_status=lambda **kargs: kwargs,
        iter_lines=lambda **kwargs: data,
        headers={'Content-Type': "text/plain"}
    )


@pytest.fixture()
def mock_metrics_request():
    with mock.patch('requests.get', new=mock_get):
        yield
```

In this iteration, without `raise_for_status` included, an error got raised from `datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py:586`

Reply:
Got it, thanks for the explanation 🙇